tests/api/routes/test_proposals.py

```diff
@@ -95,3 +98,59 @@ def test_create_proposal(client, reverse_url):
     liked_by = created_proposal.liked_by.all()
     assert_that(liked_by, has_length(1))
     assert_that(liked_by.first().id, equal_to(fanboy.id))
+
+
+@freeze_time("2022-10-26 23:23:23")
+@pytest.mark.django_db
+def test_like_proposal(client):
+    subject = "anything"
+    password = "Password10!"
+    description = "Any description"
+    proposer = UserBuilder().with_email("a@a.com").build()
+    fanboy = UserBuilder().with_email("b@b.com").with_password(password).build()
+    topic = TopicBuilder().build()
+    proposal = (
+        ProposalBuilder()
+        .with_subject(subject)
+        .with_description(description)
+        .with_topics([topic])
+        .with_proposed_by(proposer)
+        .build()
+    )
+    url = reverse_lazy("api-alpha:like_proposal", kwargs={"proposal_id": proposal.id})
+    client.login(username=fanboy.email, password=password)
+
+    response = client.put(url, data={}, content_type="application/json")
+
+    assert_that(response.status_code, equal_to(HTTPStatus.NO_CONTENT))
+    liked_by = proposal.liked_by.all()
+    assert_that(liked_by, has_length(1))
+    assert_that(liked_by.first().id, equal_to(fanboy.id))
+
+
+@freeze_time("2022-10-26 23:23:23")
+@pytest.mark.django_db
+def test_unlike_proposal(client):
+    subject = "anything"
+    password = "Password10!"
+    description = "Any description"
+    proposer = UserBuilder().with_email("a@a.com").build()
+    fanboy = UserBuilder().with_email("b@b.com").with_password(password).build()
+    topic = TopicBuilder().build()
+    proposal = (
+        ProposalBuilder()
+        .with_subject(subject)
+        .with_description(description)
+        .with_topics([topic])
+        .with_proposed_by(proposer)
+        .with_liked_by([fanboy])
+        .build()
+    )
+    url = reverse_lazy("api-alpha:like_proposal", kwargs={"proposal_id": proposal.id})
+    client.login(username=fanboy.email, password=password)
+
+    response = client.delete(url, data={}, content_type="application/json")
+
+    assert_that(response.status_code, equal_to(HTTPStatus.NO_CONTENT))
+    liked_by = proposal.liked_by.all()
```

Comment:
Add a proposal.refresh_from_db() to be confident 100%

Reply:
OK